src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,30 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
```

Comment:
The check for the login length should be extracted to a constant field with an informative name to avoid magic numbers. For example, `private static final int MIN_LOGIN_LENGTH = 6;`